packages/bugsnag_flutter/lib/src/config.dart

```diff
@@ -80,10 +80,25 @@ enum BugsnagEnabledBreadcrumbType {
 
 /// Types of telemetry that may be sent to Bugsnag for product improvement
 /// purposes.
-enum BugsnagTelemetryType {
+class BugsnagTelemetryTypes {
   /// Errors within the Bugsnag SDK.
-  internalErrors,
+  final bool internalErrors;
 
   /// Information about how Bugsnag has been configured.
-  usage,
+  final bool usage;
+
+  const BugsnagTelemetryTypes({
+    this.internalErrors = true,
+    this.usage = true,
+  });
+
+  dynamic toJson() => {
+        'internalErrors': internalErrors,
+        'usage': usage,
+      };
+
+  dynamic toList() =>
+      [if (internalErrors) "internalErrors", if (usage) "usage"];
```

Comment:
We expect the JSON to be the "wire" format here, so the `toList` implementation is what the native layers expect:

```suggestion
  dynamic toJson() => [if (internalErrors) "internalErrors", if (usage) "usage"];
```

This will make the `toList()` call in `start` redundant (as the Json encoder will automatically call `toJson`).